internal/proto/encode.go

```diff
@@ -0,0 +1,89 @@
+package proto
+
+import (
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+// WriteVersionedJSONBlob writes a JSON blob to the given writer. It expects
+// the blob to be read using 'ReadVersionedJSONBlob'.
+// A version is included via a v0 compatible hack since v0 did not include the
+// version. Specifically, the version is encoded as whitespace prefixing the
+// json data.
+func WriteVersionedJSONBlob(dst io.Writer, obj interface{}, version uint32) error {
+	versionPrefix := encodeVersion(version)
+
+	var jsonBlob bytes.Buffer
+	jsonBlob.Write(versionPrefix)
+	enc := json.NewEncoder(&jsonBlob)
+	if err := enc.Encode(obj); err != nil {
+		return err
+	}
+
+	var jsonBlobLenBuf bytes.Buffer
+	if err := binary.Write(&jsonBlobLenBuf, binary.BigEndian, int32(jsonBlob.Len())); err != nil {
+		panic(fmt.Errorf("could not binary encode an int32: %v", err))
+	}
+	if jsonBlobLenBuf.Len() != 4 {
+		panic(fmt.Errorf("int32 should be 4 bytes, not: %+v", jsonBlobLenBuf))
+	}
```

Comment:
this check seems unnecessary. it's basically a check on the go stdlib, which i don't think we need? we could just binary.Write directly to `dst`

Reply:
Yeah, you're right, I'll delete it.